app/component/itinerary/Itinerary.js

```
@@ -277,6 +280,9 @@ const Itinerary = (
   );
   const usingOwnBicycleWholeTrip =
     usingOwnBicycle && itinerary.legs.every(leg => !leg.to?.vehicleParking);
+  const usingOwnCar = itinerary.legs.some(leg => getLegMode(leg) === 'CAR');
+  const usingOwnCarWholeTrip =
+    usingOwnCar && itinerary.legs.every(leg => !leg.to?.vehicleParking);
```

can leg.to actually be null?

---

I can find checks for leg.to in many parts of the code (e.g. CarParkLeg, CarLeg, TransitLeg, ViaLeg, WalkLeg). I assume therefore there is a chance for it to be null. Also the same code for bicycles used it in a similar way. Do you want me to look into it further or is this fine as is?

---

build/schema.graphql says in type Leg: 

 "The Place where the leg ends."
  to: Place!
! means that to cannot be null. Please fix at least your own code.

---

I removed the ? from all uses of leg.to?.field and leg.from?.field I could find. However, in legUtils removing the check caused a test to fail so I added the ? back there

---

right, unit tests often use bad mock data. Fixing code causes huge amount of test failures which is really annoying.